enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,355 @@
+2. `serviceaccount.yaml` specifies the controller's service account
+3. `05_operand_rbac.yaml` provides RBAC configuration to the CSI controller 
+4. `webhook_config.yaml`
+5. `webhook_deployment.yaml`
+6. `volumesnapshotclasses_snapshot.storage.k8s.io.yaml` defines the VolumeSnapshotClass CRD 
+7. `volumesnapshotcontents_snapshot.storage.k8s.io.yaml` defines the VolumeSnapshotContents CRD
+8. `volumesnapshots_snapshot.storage.k8s.io.yaml` defines the VolumeSnapshot CRD
+
+The controller will be deployed prior to LVMS.  Because LVMS does not depend on the controller to be running prior to its deployment,
+it is not necessary to wait for the controller to reach a ready state before starting LVMS.  Starting these components 
+concurrently will also shorten overall startup time.
+
+### Test Plan
+
+Smoke tests will be written and run as part of pre-submit tasks on MicroShift CI.  At a minimum, the tests should
+verify that a VolumeSnapshot created from an existing PVC containing data:
+
+1. results in the snapshotting of the underlying volume
+2. is addressable and accessible via a consumer PVC
+3. enables rollback of data on the original volume
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+**Images**
+
+The CSI Snapshot controller and sidecar are pieces of the Kubernetes CSI implementation.  OpenShift
+maintains downstream versions of these images and tracks them as part of OCP releases.
+
+**Manifests**
+
+MicroShift's rebase automation specifies repos in the OpenShift github organization from which specific manifests are obtained.
+For the CSI Snapshot Controller, we will specify https://github.com/openshift/cluster-csi-snapshot-controller-operator/tree/release-$RELEASE/assets
+as the remote source from which to derive the controller manifests.
+
+LVMS does not provide its manifests as plain yaml files.  Instead, these are encoded into the logic of the operator, which
+is not deployed on MicroShift.  This makes retrieving them automatically difficult.  For this reason, the LVMS manifests 
+are derived once from a running instance of the controller and stored under `microshift/assets/components/lvms/`.
+
+### Version Skew Strategy
+
+The CSI Snapshot version is tracked by the ocp-release image.  This provides sufficient guardrails
+against version skew.
+
+### Operational Aspects of API Extensions
```

Comment:
We are adding a few new APIs so we should fill in these sections about how they might fail and how someone is supposed to debug the failures when something goes wrong. Are there specific CRs or log files to examine for errors, for example?

Can the new web hooks introduce new failure modes?

Reply:
👍🏻